src/net_processing.cpp

```diff
@@ -3345,6 +3359,33 @@ void PeerManagerImpl::ProcessCompactBlockTxns(CNode& pfrom, Peer& peer, const Bl
     return;
 }
 
+void PeerManagerImpl::LogBlockHeader(const uint256 hash, const std::optional<int> height, const CNode& peer, const bool via_compact_block, const bool via_unsolicited_block) {
+    // To prevent log spam, this function should only be called after it was determined that a
+    // header is both new and valid.
+    //
+    // These messages are valuable for detecting potential selfish mining behavior;
+    // if multiple displacing headers are seen near simultaneously across many
+    // nodes in the network, this might be an indication of selfish mining.
```

Comment:
I'm not sure if it's precisely defined anywhere, but the term is used to describe a situation where a miner intentionally doesn't broadcast their new block and instead waits for a competitor to mine at the same height, hoping that the network picks their side of the fork. When @jamesob first opened an earlier version of this PR, there were rumors of such an attack happening, but this may have been a misunderstanding based on our block connection logging.
https://bitcoin.stackexchange.com/questions/tagged/selfish-mining